llvm-hs/src/LLVM/Internal/ObjectFile.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module LLVM.Internal.ObjectFile (
+    ObjectFile,
+    createObjectFile,
+    disposeObjectFile
+  ) where
+
+import Control.Monad.IO.Class
+import Control.Monad.AnyCont
+import Foreign.Ptr
+
+import LLVM.Prelude
+import LLVM.Internal.Coding
+import LLVM.Internal.MemoryBuffer
+import qualified LLVM.Internal.FFI.ObjectFile as FFI
+
+data ObjectFile = ObjectFile (Ptr FFI.ObjectFile)
```

Comment:
Can you make this a newtype? I don’t think we gain anything from the extra indirection here.